### PR TITLE
Add new APIs (two List APIs) and update examples

### PR DESCRIPTION
**Pull request by bashahee**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by bashahee** [comment]
@jaredmoo can you please review?

**Review by jaredmoo — APPROVED** [review]
Approved assuming that this is available in at least 1 production region.

**Comment by bashahee** [comment]
@amarzavery Can you please approve and merge?

**Comment by bashahee** [comment]
@jaredmoo can this PR be merged?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/DatabaseBlobAuditingCreateMax.json
Comment (amarzavery) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/DatabaseBlobAuditingCreateMax.json]: Please make sure that the json formatting is set to 2 spaces every where

**Review by amarzavery — APPROVED** [review]

**Comment by bashahee** [comment]
@amarzavery can this PR be merged?

**Comment by NullMDR** [comment]
@openapibot sdkautomation rebuild

**Comment by bashahee** [comment]
@amarzavery ping

**Comment by bashahee** [comment]
@zikalino Can this PR be merged?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json
Comment (KrisBash) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json]: Each resource type should have a collection/list API and a point/get-by-name API. Is this a singleton where the collection would return a single item? Or are there numerous items that can be returned?  Using a singleton as example, we should have both:
'/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/auditingSettings'
and
'/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/auditingSettings]/default'
paths

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/DatabaseAuditingSettingsList.json
Comment (KrisBash) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/DatabaseAuditingSettingsList.json]: This id is wrong, it doesn't have the auditingSettings or resource name in the ID

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json
Comment (KrisBash) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json]: The PR title mentions two List APIs. Is one missing?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json
Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json]: This is a singleton, a collection of one item is returned.
The `GET` API is described by `"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/auditingSettings/{blobAuditingPolicyName}"`, while the `LIST` API is described by `"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/auditingSettings"`

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json
Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json]: No, one API works on a database and is present in above file. The other one works on a server, and is placed under 2017-03-01-preview/blobAuditing.json.

**Comment by bashahee** [comment]
@KrisBash Can you please take a look?

**Comment by yungezz** [comment]
Hi @KrisBash could you pls look at the PR again after comments address? thanks.

**Comment by bashahee** [comment]
@KrisBash @yungezz Can please merge this PR?
It was opened long time ago, and comments were addressed. Moreover, these APIs are in production, and they need to be properly documented.

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
Hi @KrisBash could  you pls change `changes requested` review? Otherwise I cannot merge this one. thanks.

**Pull request merged**
